workflow/executor/executor.go

```diff
@@ -724,19 +715,25 @@ func (we *WorkflowExecutor) CaptureScriptResult(ctx context.Context) error {
 // AnnotateOutputs annotation to the pod indicating all the outputs.
 func (we *WorkflowExecutor) AnnotateOutputs(ctx context.Context, logArt *wfv1.Artifact) error {
 	outputs := we.Template.Outputs.DeepCopy()
+	containerName := common.MainContainerName
 	if logArt != nil {
+		containerName = strings.TrimSuffix(logArt.Name, "-logs")
 		outputs.Artifacts = append(outputs.Artifacts, *logArt)
 	}
 
 	if !outputs.HasOutputs() {
 		return nil
 	}
-	log.Infof("Annotating pod with output")
+
+	key := fmt.Sprintf("%s-%s", common.AnnotationKeyOutputs, containerName)
 	outputBytes, err := json.Marshal(outputs)
 	if err != nil {
+		log.Infof("in error: %s", err.Error())
```

Comment:
always do structured logging - so use log.WithError, never use Infof

Reply:
removed this debug log completely